thirdeye/thirdeye-frontend/app/pods/services/rootcause-fetcher/service.js

```diff
@@ -0,0 +1,210 @@
+import Service from '@ember/service';
+import { get, getProperties, set, setProperties } from '@ember/object';
+import fetch from 'fetch';
+import _ from 'lodash';
+import RSVP from 'rsvp';
+
+/**
+ * Comparator for of MyPromise for priority queue
+ */
+const ROOTCAUSE_FETCHER_COMPARATOR = (a, b) => {
+  if (a.priority !== b.priority) { return a.priority < b.priority; }
+  if (a.externalOrder !== b.externalOrder) { return a.externalOrder < b.externalOrder; }
+  if (a.internalOrder !== b.internalOrder) { return a.internalOrder < b.internalOrder; }
+  return true;
+};
+
+//
+// FROM : https://stackoverflow.com/questions/42919469/efficient-way-to-implement-priority-queue-in-javascript
+//
+/**
+ * Priority Queue via max-heap
+ */
+const top = 0;
+const parent = i => ((i + 1) >>> 1) - 1;
+const left = i => (i << 1) + 1;
+const right = i => (i + 1) << 1;
+
+class PriorityQueue {
```

Comment:
There a need for document for both Classes and also what this new service will deliver. It's good practice on all services and utils etc. So we know how we can use it elsewhere. A link to `stackoverflow` might always give us enough insight. Thanks.

Reply:
ok added more JS doc